src/components/ImageUploader/src/ImageSelection.vue

```diff
@@ -0,0 +1,174 @@
+<template>
+	<div
+		:class="{
+			[$s.ImageSelectionContainer]: true,
+			[$s.ImageSelectionContainerError]: hasError
+		}"
+		role="img"
+		:style="bgImageStyle"
+	>
+		<div
+			v-if="isUploading"
+			:class="$s.ImageSelectionLoaderContainer"
+		>
+			<loading :class="$s.ImageSelectionLoader" />
+		</div>
+
+		<progress-bar
+			:class="$s.ImageSelectionProgressContainer"
+			:style="progressContainerStyle"
+			:progress="image.progress"
+			size="xsmall"
+		/>
+
+		<button
+			v-if="!isUploading"
+			type="button"
+			:class="$s.ImageSelectionRemoveButton"
+			@click="$emit('removeImage', image.id)"
+		>
+			<x-icon
+				inline
+				:class="$s.ImageSelectionRemoveIcon"
+			/>
+		</button>
+	</div>
+</template>
+
+<script>
+import { MLoading as Loading } from '@square/maker/components/Loading';
+import { MProgressBar as ProgressBar } from '@square/maker/components/ProgressBar';
+import XIcon from '@square/maker-icons/X';
+import { IMAGE_SELECTOR_STATUSES } from './constants';
+
+const MAX_PROGRESS = 100;
+const PROGRESS_OPACITY = { shown: 1, hidden: 0 };
+
+export default {
+	name: 'MImageSelection',
+
+	components: {
+		Loading,
+		ProgressBar,
+		XIcon,
+	},
+
+	props: {
+		image: {
+			type: Object,
+			required: true,
+		},
+	},
+
+	computed: {
+		bgImageStyle() {
+			const url = `url("${this.image.url}")`;
+			const loadingGradient = 'linear-gradient(0deg, rgba(255, 255, 255, 0.5), rgba(255, 255, 255, 0.5))';
+			const loadedGradient = 'linear-gradient(180deg, rgba(0, 0, 0, 0.5) 0%, rgba(0, 0, 0, 0) 50%)';
```

Comment:
We have a loading component that could replace this. Should simplify this code a bit.

https://github.com/square/maker/tree/master/src/components/Loading

Reply:
You can see how it's used in Button for example: https://github.com/square/maker/blob/master/src/components/Button/src/Button.vue#L20-L23

Reply:
[We do actually use `MLoader` here](https://github.com/square/maker/pull/137/files/e3d8eceac94d7572e35a343cb2c012370dbe44fc#diff-9336009bb13b843a4ee64eefba32b0cd7682e64580c2c810c6e22e51abb1148aR14), this is meant as an overlay for the component while it is being uploaded.

Reply:
Should this be a new Maker component? `MOverlay`?

Reply:
Interesting idea. I'll run it by Shimaa. Though I think that could come as a follow up.

Reply:
This is mostly an optimization, but I think between the existing loading / progress bar components a lot of these styles can go away. The overlay is an interesting additional component, but not that much of an optimization over the additional code to show the loading state of an image. This isn't necessarily a blocker to move forward though. We can either do it now or later.